src/components/application_manager/test/message_helper/message_helper_test.cc

```diff
@@ -957,6 +976,422 @@ TEST_F(MessageHelperTest, SubscribeApplicationToSoftButton_CallFromApp) {
       message_params, appSharedPtr, function_id);
 }
 
+TEST_F(MessageHelperTest, SendOnAppUnregNotificationToHMI) {
+  const bool kIsUnexpectedDisconnect = false;
+
+  MockApplicationSharedPtr mock_app = ::utils::MakeShared<AppMock>();
+
+  uint32_t mock_hmi_app_id = 1u;
```

Comment:
@vlantonov I don't think Return somehow changes the variable. 
[Gtest CheatSheet](https://github.com/google/googletest/blob/master/googlemock/docs/CheatSheet.md)
`Return value. If the type of value is different to the mock function's return type, value is converted to the latter type at the time the expectation is set, not when the action is executed.`